### PR TITLE
#4864: use tree walker for replace text brick

### PR DESCRIPTION
**Pull request by twschiller**
## What does this PR do?

- Follow up to #4864
- Use TreeWalker instead of recursion

## Checklist

- [x] Add tests: N/A
- [x] Run Storybook and manually confirm that all stories are working: N/A
- [X] Designate a primary reviewer: @fregante

### CODE REVIEW

File: src/blocks/effects/replaceText.ts [hunk 1]
```diff
@@ -33,21 +33,26 @@ export function replaceText({
...
 
-  if (node.nodeType === Node.TEXT_NODE) {
-    node.textContent = node.textContent.replaceAll(pattern, replacement);
-    return;
-  }
+  for (
+    let currentNode = walker.nextNode();
+    currentNode;
+    currentNode = walker.nextNode()
+  ) {
```
Comment (fregante) [inline src/blocks/effects/replaceText.ts:44]: I'd probably go for this instead a `for` loop
```suggestion
	let currentNode;
  // eslint-disable-next-line I'm sure there's a rule preventing this
  while (currentNode = walker.nextNode()) {
```

Other than that, it looks fine.

**Review by fregante — APPROVED** [review]

File: src/blocks/effects/replaceText.ts (hunk 1 above)
Comment (twschiller) [reply src/blocks/effects/replaceText.ts:44]: Copilot wrote it, so I'm not particularly attached to it. There is indeed an eslint rule no-cond-assign for that condition. I agree it looks slightly cleaner though

Comment (twschiller) [reply src/blocks/effects/replaceText.ts]: Prettier magically eliminated the lint warning -- there's no lint if you wrap the whole expression in parentheses 🤷

Comment (fregante) [reply src/blocks/effects/replaceText.ts]: # BOTS GIVETH
# BOTS TAKETH

**Pull request merged**